core/server/api/webhooks.js

```diff
@@ -0,0 +1,155 @@
+// # Webhooks API
+// RESTful API for creating webhooks
+// also known as "REST Hooks", see http://resthooks.org
+var Promise = require('bluebird'),
+    _ = require('lodash'),
+    https = require('https'),
+    url = require('url'),
+    pipeline = require('../utils/pipeline'),
+    apiUtils = require('./utils'),
+    models = require('../models'),
+    errors = require('../errors'),
+    logging = require('../logging'),
+    i18n = require('../i18n'),
+    docName = 'webhooks',
+    webhooks;
+
+// TODO: Use the request util. Do we want retries here?
```

Comment:
Here is a TODO left. Was your plan to tidy that up later?

Reply:
Yeah, this one can be tidied up later